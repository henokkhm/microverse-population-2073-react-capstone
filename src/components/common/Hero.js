import PropTypes from 'prop-types';

import styles from '../../styles/Hero.module.css';

function Hero({
  title,
  populationIn2073,
  imgURL,
  imgAlt,
  slantImage,
}) {
  const imageStyles = slantImage ? { transform: 'rotateZ(-18deg)' } : {};

  return (
    <section className={styles.hero}>
      <div className={styles.heroImgWrapper}>
        <img style={imageStyles} className={styles.heroImg} src={imgURL} alt={imgAlt} />
      </div>
      <div className={styles.heroContent}>
        <h2 className={styles.heroTitle}>{title}</h2>
        <p className={styles.heroSubtitle}>
          <span>{populationIn2073.toLocaleString()}</span>
          {' '}
          <span>people</span>
        </p>
      </div>
    </section>
  );
}

Hero.defaultProps = {
  slantImage: false,
};

Hero.propTypes = {
  title: PropTypes.string.isRequired,
  populationIn2073: PropTypes.number.isRequired,
  imgURL: PropTypes.string.isRequired,
  imgAlt: PropTypes.string.isRequired,
  slantImage: PropTypes.bool,
};

export default Hero;
