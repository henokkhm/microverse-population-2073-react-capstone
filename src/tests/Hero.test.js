import { render } from '@testing-library/react';

import Hero from '../components/common/Hero';

describe('The Hero component', () => {
  it('matches snapshot', () => {
    const { title, worldImageURL, worldPopulationIn2073 } = {
      title: 'World [mock]',
      worldImageURL: '/images/globe.gif',
      worldPopulationIn2073: 10337756759,
    };

    const { asFragment } = render(
      <Hero
        title={title}
        imgURL={worldImageURL}
        imgAlt="image of the glob"
        populationIn2073={worldPopulationIn2073}
        slantImage
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
