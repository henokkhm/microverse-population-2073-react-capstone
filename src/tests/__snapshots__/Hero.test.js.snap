// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Hero component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="hero"
  >
    <div
      class="heroImgWrapper"
    >
      <img
        alt="image of the glob"
        class="heroImg"
        src="/images/globe.gif"
        style="transform: rotateZ(-18deg);"
      />
    </div>
    <div
      class="heroContent"
    >
      <h2
        class="heroTitle"
      >
        World [mock]
      </h2>
      <p
        class="heroSubtitle"
      >
        <span>
          10,337,756,759
        </span>
         
        <span>
          people
        </span>
      </p>
    </div>
  </section>
</DocumentFragment>
`;
